src/main/groovy/com/vanniktech/android/javadoc/Generation.groovy

```diff
@@ -25,29 +29,61 @@ class Generation implements Plugin<Project> {
 
         project.extensions.create("androidJavadoc", AndroidJavadocExtension)
 
-        project.allprojects {
-            afterEvaluate { Project p ->
-                // Do not throw exception anymore. We want to support big projects with a lot of modules that not all of them are android.
-                if (p.hasProperty('android')) {
-                    applyPluginToProject(p);
-                } else {
-                    logger.info "${p.name} is not an android project - plugin is not applied"
+        if (isRoot(project)) {
+            project.allprojects { Project p ->
+                applyPlugin(p)
+            }
+        } else {
+            applyPlugin(project)
+        }
+    }
+
+    private void applyPlugin(Project project) {
+        if (project.hasProperty('android')) {
+            applyPluginToProject(project)
+        } else {
+            // Waiting for android plugin to be applied
+            project.plugins.whenPluginAdded { Plugin plugin ->
+                if (project.plugins.hasPlugin("com.android.library")
```

Comment:
I have replaced this by `plugin instanceOf com.android.build.gradle.BasePlugin`
It cover all other android plugins. 

Also, there was an issue because plugin was applied each time a new plugin was applied after android ones. Now Javadoc plugin is applied only once.